modules/tinymce/src/themes/silver/main/ts/ui/window/SilverDialogCommon.ts

```diff
@@ -70,37 +69,57 @@ const getEventExtras = (lazyDialog: () => AlloyComponent, providers: UiFactoryBa
   }
 });
 
-const renderModalDialog = <T>(spec: DialogSpec, initialData: T, dialogEvents: AlloyEvents.AlloyEventKeyAndHandler<any>[], backstage: UiFactoryBackstage): AlloyComponent => {
-  const updateState = (_comp: AlloyComponent, incoming: T) => Optional.some(incoming);
-
-  return GuiFactory.build(Dialogs.renderDialog({
-    ...spec,
-    firstTabstop: 1,
-    lazySink: backstage.shared.getSink,
-    extraBehaviours: [
-      // Because this doesn't define `renderComponents`, all this does is update the state.
-      // We use the state for the initialData. The other parts (body etc.) render the
-      // components based on what reflecting receives.
-      Reflecting.config({
-        channel: `${dialogChannel}-${spec.id}`,
-        updateState,
-        initialData
-      }),
-      RepresentingConfigs.memory({ }),
-      ...spec.extraBehaviours
-    ],
-    onEscape: (comp) => {
-      AlloyTriggers.emit(comp, formCancelEvent);
-    },
-    dialogEvents,
-    eventOrder: {
-      [SystemEvents.receive()]: [ Reflecting.name(), Receiving.name() ],
-      [SystemEvents.attachedToDom()]: [ 'scroll-lock', Reflecting.name(), 'messages', 'dialog-events', 'alloy.base.behaviour' ],
-      [SystemEvents.detachedFromDom()]: [ 'alloy.base.behaviour', 'dialog-events', 'messages', Reflecting.name(), 'scroll-lock' ]
-    }
-  }));
+const fullscreenClass = 'tox-dialog--fullscreen';
+const largeDialogClass = 'tox-dialog--width-lg';
+const mediumDialogClass = 'tox-dialog--width-md';
+
+const getDialogSizeClass = (size: Dialog.DialogSize): Optional<string> => {
+  switch (size) {
+    case 'large':
+      return Optional.some(largeDialogClass);
+    case 'medium':
+      return Optional.some(mediumDialogClass);
+    default:
+      return Optional.none();
+  }
+};
+
+const updateDialogSizeClass = (size: Dialog.DialogSize, component: AlloyComponent): void => {
+  const sugarBody = SugarElement.fromDom(component.element.dom);
+  const classes = Classes.get(sugarBody);
+  if (!Arr.contains(classes, fullscreenClass)) {
+    const currentSizeClass = Arr.find(classes, (c) => c === largeDialogClass || c === mediumDialogClass);
+    currentSizeClass.map((c) => Classes.remove(sugarBody, [ c ]));
+    getDialogSizeClass(size).map((dialogSizeClass) => Classes.add(sugarBody, [ dialogSizeClass ]));
```

Comment:
Sugar puts singular and plural in different places - in fact the plural generally delegates to the singular - so this can be done far more easily
(also please don't use `map` with functions that don't return a value, it makes a new `Optional<undefined>` that is then ignored)
```suggestion
    currentSizeClass.each((c) => Class.remove(sugarBody, c));
    getDialogSizeClass(size).each((dialogSizeClass) => Class.add(sugarBody, dialogSizeClass));
```